tests/playwright/src/specs/kubernetes-context-smoke.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { PreferencesPage } from '../model/pages/preferences-page';
+import { expect as playExpect, test } from '../utility/fixtures';
+
+const testContexts = ['context-1', 'context-2', 'context-3'];
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+test.beforeAll(async ({ runner, welcomePage }) => {
+  runner.setVideoAndTraceName('kube-context-e2e');
+
+  // copy testing kubeconfig file to the expected location
+  const kubeConfigPathSrc = path.resolve(__dirname, '..', '..', 'resources', 'test-kube-config');
+  const kubeConfigPathDst = path.resolve(__dirname, '..', '..', 'resources', 'kube-config');
+  fs.copyFileSync(kubeConfigPathSrc, kubeConfigPathDst);
+
+  await welcomePage.handleWelcomePage(true);
+});
+
+test.afterAll(async ({ runner }) => {
+  test.setTimeout(120000);
+  await runner.close();
+});
+
+test.describe.serial('Verification of kube context management @smoke', () => {
+  test('Load custom kubeconfig in Preferences', async ({ navigationBar }) => {
+    // open preferences page
+    const settingsBar = await navigationBar.openSettings();
+    await settingsBar.expandPreferencesTab();
+    const preferencesPage = await settingsBar.openTabPage(PreferencesPage);
+    await playExpect(preferencesPage.heading).toBeVisible();
+
+    const kubeConfigPathDst = path.resolve(__dirname, '..', '..', 'resources', 'kube-config');
+    await preferencesPage.selectKubeFile(kubeConfigPathDst);
+    await playExpect(preferencesPage.heading).toBeVisible();
+  });
+
+  test('Can load kube contexts in Kubernetes page', async ({ navigationBar }) => {
+    const settingsBar = await navigationBar.openSettings();
+    const kubePage = await settingsBar.openTabPage(KubeContextPage);
+    await playExpect(kubePage.heading).toBeVisible();
+
+    await playExpect.poll(async () => await kubePage.pageIsEmpty(), { timeout: 10000 }).toBeFalsy();
+    for (const context of testContexts) {
+      const row = await kubePage.getContextRowByName(context);
+      if (row === undefined) {
+        throw Error(`Context: '${context}' not found`);
+      }
+      await playExpect(row.getByLabel('Context Name', { exact: true })).toHaveText(context);
+      await playExpect(row.getByLabel('Context Cluster')).toHaveText(context + '-cluster');
+      await playExpect(row.getByLabel('Context Server')).toHaveText(context + '-server');
+      await playExpect(row.getByLabel('Context User')).toHaveText(context + '-user');
+
+      if (context === 'context-1') {
+        // check if the context is default
+        await playExpect.poll(async () => await kubePage.isContextDefault(context), { timeout: 10000 }).toBeTruthy();
+        await playExpect(row.getByLabel('Set as Current Context')).not.toBeVisible();
+        await playExpect(row.getByLabel('Context Namespace')).toHaveText(context + '-namespace');
+      } else {
+        await playExpect(row.getByLabel('Set as Current Context')).toBeVisible();
```

Comment:
All of these properties should be exposed from the relevant page object model via methods that accept the context name as a parameter because we might want to check these fields in other tests too not to mention and doing it like this is not scalable. 

For example for `Context Cluster` it should be something like `await kubePage.getContextCluster(context)` which returns a string type.

This way we abstract the looking for the row in question and also the getByLabel() method in the POM and if/when the locators will change we will only have to change them in the POM, not individually in every test that might use them.